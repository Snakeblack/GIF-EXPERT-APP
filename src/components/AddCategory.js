import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('Buscar GIF');

    const handleInputChange = ( e ) => {
        // console.log(e.target.value);
        setInputValue( e.target.value );
    }

    const handleSubmit = ( e ) => {
        e.preventDefault();
        // console.log( 'Submit hecho' );
        setCategories( cats => [...cats, inputValue] );
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.string
}
