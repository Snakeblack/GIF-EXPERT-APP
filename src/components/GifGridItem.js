import React from 'react';
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {

    //console.log( id, title, url );
    const titleLength = title.indexOf('GIF');
    const titleShort = title.substring( 0, titleLength ).trim();


    return (
        <div className='card animate__animated animate__fadeIn'>
            <div className='img'>
                <img src={ url } alt={ titleShort } />
                <p> { titleShort } </p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}