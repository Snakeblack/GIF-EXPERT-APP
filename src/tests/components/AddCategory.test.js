import React from "react";
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";


describe('Pruebas en <AddCategory.js />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/>);

    beforeEach( () => {

        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/>);

    })

    test('Debe de mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })

    // test('Debe de cambiar la caja de texto', () => {
        
    //     const input = wrapper.find('input');
    //     const value = 'Test';
        
    //     input.simulate('change', { target: { value } });

    //     expect( wrapper.find('p').text().trim() ).toBe( value );

    // })
    
    test('NO debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });

        expect( setCategories ).not.toHaveBeenCalled();

    })
    


})