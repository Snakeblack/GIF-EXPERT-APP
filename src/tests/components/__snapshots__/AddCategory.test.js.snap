// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory.js /> Debe de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="animate__animated animate__fadeIn"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
